### PR TITLE
create legacy ps related services only when num_ps_pods==0

### LINKED ISSUE #1370: Master uses legacy PS only when `num_ps_pods==0`
When `num_ps_pods == 0`, the master uses the legacy PS to create `MasterServicer`, `embedding_service`, `evaluation_service`, `checkpoint_service`.

For `num_ps_pods > 0 > 0`, the master will use the new PS design and create PS pods.

The old `MasterServicer` and `embedding_service` will be implemented in PS pods. 
`evaluation_service` and `checkpoint_service` will be redesigned in other projects.

Note that the legacy PS codes will be removed after we have implemented the new PS.

### PR DESCRIPTION
**Pull request by skydoorkai**
fix #1370

### CODE REVIEW

File: elasticdl/python/master/master.py [hunk 1]
```diff
@@ -104,21 +105,33 @@ def __init__(self, args):
         )
         self.optimizer = self.model_module[args.optimizer]()
 
-        # Initialize checkpoint service
-        self.checkpoint_service = self._create_checkpoint_service(args)
+        # checkpoint_service, evaluation_service and embedding_service will be
```
Comment (mhaoli) [inline elasticdl/python/master/master.py]: Do we need to mark this line as a TODO?

File: elasticdl/python/master/master.py [hunk 2]
```diff
@@ -104,21 +105,33 @@ def __init__(self, args):
...
+        # redesigned after distributed PS is implemented
+        if self.num_ps_pods:
+            self.checkpoint_service = None
+            self.evaluation_service = None
+            self.self.embedding_service_endpoint = None
```
Comment (mhaoli) [inline elasticdl/python/master/master.py]: duplicated `self`

File: elasticdl/python/master/master.py (hunk 1 above)
Comment (skydoorkai) [reply elasticdl/python/master/master.py]: added.

File: elasticdl/python/master/master.py (hunk 2 above)
Comment (skydoorkai) [reply elasticdl/python/master/master.py]: fixed.

**Review by mhaoli — APPROVED** [review]

**Pull request merged**